pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -459,7 +459,8 @@ let
     )
     ++ setupHaskellDepends
     ++ collectedToolDepends
-    ++ optional stdenv.hostPlatform.isGhcjs nodejs;
+    ++ optional stdenv.hostPlatform.isGhcjs nodejs
+    ++ optional (ghc.useLdLld or false) buildPackages.llvmPackages.bintools;
```

Comment:
This shouldn't be necessary. The GHC settings file can hold the absolute path and GHC should use that.

Reply:
As it stands if remove this line then linking Haskell packages fails. I'm not sure why. If we pick this back up in the future I'll look into it